src/vfio_device.rs

```diff
@@ -21,7 +21,7 @@ use kvm_bindings::{
 use kvm_ioctls::DeviceFd;
 use log::{debug, error, warn};
 use vfio_bindings::bindings::vfio::*;
-use vm_memory::{Address, GuestMemory, GuestMemoryRegion, MemoryRegionAddress};
+use vm_memory::{Address, GuestMemory, GuestMemoryRegion, GuestMemoryMmap, GuestRegionMmap, MemoryRegionAddress};
```

Comment:
We should use generic type <M: GuestMemory> instead of concrete GuestMemoryMmap here.

Reply:
In Cloud Hypervisor, we switched to use mem: GuestMemoryAtomic< GuestMemoryMmap > to save memory, we found Cloud Hypervisor will have build error if use generic type <M: GuestMemory> in vfio-ioctls:
```
.setup_dma_map(&self.mem.memory())
    ^^^^^^^^^^ the trait `vm_memory::GuestMemory` is not implemented for `vm_memory::GuestMemoryLoadGuard<vm_memory::GuestMemoryMmap>`
```
Any suggestions?

Reply:
Could you please try .setup_dma_map(self.mem.memory().deref())?

Reply:
it works, thanks :-)